python/tests/test_events.py

```diff
@@ -0,0 +1,366 @@
+        # uncomment `to_file` to save SBML model to file for inspection
+        # to_file=sbml_test_models / (model_name + '.sbml'),
+    )
+
+    # AMICI model
+    amici_model = create_amici_model(
+        sbml_model=sbml_model,
+        model_name=request.param,
+    )
+    amici_model.setTimepoints(timepoints)
+
+    return amici_model, parameters, timepoints, x_pected, sx_pected
+
+
+def test_models(model):
+    amici_model, parameters, timepoints, x_pected, sx_pected = model
+
+    result_expected_x = np.array([
+        x_pected(t, **parameters)
+        for t in timepoints
+    ])
+    result_expected_sx = np.array([
+        sx_pected(t, parameters)
+        for t in timepoints
+    ])
+
+    # --- Test the state trajectories without sensitivities -------------------
+    # Does the AMICI simulation match the analytical solution?
+    solver = amici_model.getSolver()
+    solver.setAbsoluteTolerance(1e-15)
+    rdata = runAmiciSimulation(amici_model, solver=solver)
+    np.testing.assert_almost_equal(rdata['x'], result_expected_x, decimal=5)
+
+    # Show that we can do arbitrary precision here (test 8 digits)
+    solver = amici_model.getSolver()
+    solver.setAbsoluteTolerance(1e-15)
+    solver.setRelativeTolerance(1e-12)
+    rdata = runAmiciSimulation(amici_model, solver=solver)
+    np.testing.assert_almost_equal(rdata['x'], result_expected_x, decimal=8)
+
+    # --- Test the state trajectories with sensitivities ----------------------
+    # Show that we can do arbitrary precision here (test 8 digits)
+    solver = amici_model.getSolver()
+    solver.setSensitivityOrder(SensitivityOrder.first)
+    solver.setSensitivityMethod(SensitivityMethod.forward)
+    rdata = runAmiciSimulation(amici_model, solver=solver)
+    np.testing.assert_almost_equal(rdata['x'], result_expected_x, decimal=5)
+    np.testing.assert_almost_equal(rdata['sx'], result_expected_sx, decimal=5)
+
+    # Show that we can do arbitrary precision here (test 8 digits)
+    solver = amici_model.getSolver()
+    solver.setSensitivityOrder(SensitivityOrder.first)
+    solver.setSensitivityMethod(SensitivityMethod.forward)
+    solver.setAbsoluteTolerance(1e-15)
+    solver.setRelativeTolerance(1e-13)
+    solver.setAbsoluteToleranceFSA(1e-15)
+    solver.setRelativeToleranceFSA(1e-13)
+    rdata = runAmiciSimulation(amici_model, solver=solver)
+    np.testing.assert_almost_equal(rdata['x'], result_expected_x, decimal=8)
+    np.testing.assert_almost_equal(rdata['sx'], result_expected_sx, decimal=8)
+
+
+def create_amici_model(sbml_model, model_name) -> AmiciModel:
+    sbml_importer = SbmlImporter(sbml_model)
+    output_dir = sbml_test_models_output_dir / model_name
+    sbml_importer.sbml2amici(
+        model_name=model_name,
+        output_dir=str(output_dir)
+    )
+
+    sys.path.insert(0, str(output_dir.resolve()))
+    model_module = importlib.import_module(model_name)
+
+    model = model_module.getModel()
+    return model
+
+
+def create_sbml_model(
```

Comment:
I appreciate the time and effort you put into this, but what about using `https://simplesbml.readthedocs.io/en/latest/` instead of reinventing the wheel?

Reply:
That function was just cut-copy-pasted from the Heaviside test-case... :D
Doesn't make things better, true. But Here I really didn't put much effort. Can slim things down though by using simplesbml.

Reply:
if its a simple copy paste, please import instead. Not necessary to replace this now, but maybe you can open an issue that this needs to be fixed in the future?